File: PythonCode/toolkit/info_divergence.py
import numpy as np


def infomation_divergence(sigma_a, sigma_b):
    """
    Input:
    sigma_a: the covariance matrix Sigma of network a
    sigma_b: the covariance matrix Sigma of network b

    Output:
    d_ab: the information divergence from a to b
    d_ba: the information divergence from b to a
    """

    # d_ab = 0.5*(np.trace(np.matmul(np.linalg.inv(sigma_b), sigma_a)) - sigma_a.shape[0] 
    #             + np.log(np.linalg.det(sigma_b)/np.linalg.det(sigma_a)))
    d_ab = 0.5*(np.trace(np.matmul(np.linalg.inv(sigma_b), sigma_a)) - sigma_a.shape[0] 
                + np.log(np.prod(np.linalg.eigvals(sigma_b)/np.linalg.eigvals(sigma_a))))
    # d_ba = 0.5*(np.trace(np.matmul(np.linalg.inv(sigma_a), sigma_b)) - sigma_b.shape[0] 
    #             + np.log(np.linalg.det(sigma_a)/np.linalg.det(sigma_b)))
    d_ba = 0.5*(np.trace(np.matmul(np.linalg.inv(sigma_a), sigma_b)) - sigma_b.shape[0] 
                + np.log(np.prod(np.linalg.eigvals(sigma_a)/np.linalg.eigvals(sigma_b))))
    return d_ab, d_ba

File: MatlabCode/InfoDivergence.m
function [DAB,DBA]=InfoDivergence(NSigmaA,NSigmaB)
% Input: 
% NSigmaA: the covariance matrix Sigma of network A
% NSigmaB: the covariance matrix Sigma of network B

% Output:
% DAB: the information divergence from network A to network B
% DBA: the information divergence from network B to network A
% Note that DAB may be different from DBA. Information divergence is 
% directional

%% Calculate DAB 
DAB=real(0.5*(trace(NSigmaA/NSigmaB)-size(NSigmaA,1)+log(prod(eig(NSigmaB)./eig(NSigmaA)))));
%% Calculate DBA
DBA=real(0.5*(trace(NSigmaB/NSigmaA)-size(NSigmaB,1)+log(prod(eig(NSigmaA)./eig(NSigmaB)))));
